### PR TITLE
Changes as per kubernetes client 5.12.2

### PR DESCRIPTION
**Pull request by mukeshkumarkulmi**

### CODE REVIEW

File: server/src/main/scala/org/apache/livy/utils/SparkKubernetesApp.scala [hunk 1]
```diff
@@ -553,8 +554,6 @@ private[utils] object KubernetesExtensions {
         .addNewPath()
         .withPath(s"/$appTag/?(.*)")
         .withNewBackend()
-        .withServiceName(service.getMetadata.getName)
```
Comment (epheatt) [inline server/src/main/scala/org/apache/livy/utils/SparkKubernetesApp.scala:556]: @mukeshkumarkulmi Should this read as follows to preserve the access 

```
        .withNewBackend()
        .withNewService()
        .withName(service.getMetadata.getName)
        .withPort(service.getSpec.getPorts.get(0).getName)
        .endBackend()

Comment (epheatt) [inline server/src/main/scala/org/apache/livy/utils/SparkKubernetesApp.scala:556]: @mukeshkumarkulmi Should this read as follows to preserve the access 

```
        .withNewBackend()
        .withNewService()
        .withName(service.getMetadata.getName)
        .withNewPort()
        .withName(service.getSpec.getPorts.get(0).getName)
        .endPort()
        .endService()
        .endBackend()

Comment (epheatt) [reply server/src/main/scala/org/apache/livy/utils/SparkKubernetesApp.scala:556]: Screen Shot 2022-05-03 at 5 21 09 PM

Also Tests need api changes

Comment (epheatt) [reply server/src/main/scala/org/apache/livy/utils/SparkKubernetesApp.scala:556]: Screen Shot 2022-05-03 at 5 21 09 PM

Also Tests need api changes

**Comment by mukeshkumarkulmi** [comment]
Closing this PR. Will use same branch to create PR upstream.

**Pull request closed without merging**